### PR TITLE
add status code checks

### LINKED ISSUE #50: when SG returns error, library happily returns success
Hey @gsquire, I've tried to send message and it returned Ok with the following error:
```
{"errors":[{"message":"If present, text/plain must be first, followed by text/html, followed by any other content.","field":"content","help":null}]}
```

I wonder if SG returns successful status code in this case?

### PR DESCRIPTION
**Pull request by ruseinov**
Fixes #50

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 url = "2.1"
 failure = "0.1"
+http = "0.2.1"
```
Comment (gsquire) [inline Cargo.toml:21]: Does the reqwest crate not re-export these?

**Comment by gsquire** [comment]
Thanks for the PR! Just one minor thing and I think it will be ready to merge after that.

**Comment by ruseinov** [comment]
@gsquire Sounds good, done!

**Comment by ruseinov** [comment]
I'm relatively new to Rust development, although I have been on it in many other languages, so the concept of re-export in a 3rd party package is still kinda on and off in my head, although I use it heavily in my projects.

File: Cargo.toml (hunk 1 above)
Comment (ruseinov) [reply Cargo.toml]: sure does!

**Comment by ruseinov** [comment]
@gsquire shall I bump the version as well?

**Comment by gsquire** [comment]
Nice work, @ruseinov! Thanks again.

**Comment by gsquire** [comment]
FYI I bumped the version to 0.11.0 since I would consider this a breaking change. I published it now so it should be available on https://crates.io soon.

CC @kanekv

**Pull request merged**